File: pixlus/code/addons/comp/chest.js
import { game, player } from "../lib/runtime.js"

export function place() {
	console.log("PLACED!")
}

export function interact() {
	let this_block = game.block.getBlock()

	if(this_block.type == "chest") {
		player.inventory.give("diamond", 1)
		console.log(game.block.getBlock())
		game.block.setBlock("chest:opened")
		console.log("Opened Chest")
	} else {
		console.log(game.block.getBlock())
		game.block.setBlock("chest")
		console.log("Closed Chest")
	}
}

File: pixlus/code/addons/source/chest.py
from ..lib.runtime import game, player

def place():
	print("PLACED!")

def interact():
	this_block = game.block.getBlock()

	if(this_block.type == "chest"):
		player.inventory.give("diamond", 1)
		print(game.block.getBlock())
		game.block.setBlock("chest:opened")
		print("Opened Chest")

	else:
		print(game.block.getBlock())
		game.block.setBlock("chest")
		print("Closed Chest")